src/frontend/src/components/mainWindowWithSidebar.ts

```diff
@@ -0,0 +1,127 @@
+import { html, TemplateResult } from "lit-html";
+import { ifDefined } from "lit-html/directives/if-defined.js";
+import { footer } from "./footer";
+import { icLogo } from "./icons";
+
+import { getDapps } from "$src/flows/dappsExplorer/dapps";
+import { dappsHeader } from "$src/flows/dappsExplorer/teaser";
+import { shuffleArray } from "$src/utils/utils";
+
+/** dapps visual used in sidebar */
+const dappsVisual = (): TemplateResult => {
+  const dapps = shuffleArray(getDapps());
+  return dappsHeader({
+    dapps,
+    clickable: false,
+  });
+};
+
+/**
+ * main window template
+ *
+ * To avoid having to repeat the same structure in every page,
+ * we use this component to wrap the content of each page.
+ * This way, we can change the structure of the main window
+ * in one place.
+ *
+ * It is a component that includes the logo, the footer, and the container.
+ *
+ */
+export const mainWindowWithSidebar = ({
+  slot,
+  id,
+  showFooter = true,
+  showLogo = true,
+  isWideContainer = false,
+  additionalContainerClasses = [],
+}: {
+  slot: TemplateResult;
+  id?: string;
+  showFooter?: boolean;
+  showLogo?: boolean;
+  isWideContainer?: boolean;
+  additionalContainerClasses?: string[];
+  HTMLwrapperTag?: string;
+}): TemplateResult => {
+  const containerClasses = ["l-container"];
+  if (isWideContainer === true) {
+    containerClasses.push("l-container--wide");
+  }
+  if (additionalContainerClasses.length > 0) {
+    containerClasses.push(...additionalContainerClasses);
+  }
+  return html`
+    <div class="l-wrap l-wrap--sidebar">
+      <div class="l-sidebar is-hidden--mobile">
+        <div class="l-sidebar__main">
+          <div class="c-logo c-logo--sidebar">
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            fill="none"
+            viewBox="0 0 233 111"
+          >
+            <defs>
```

Comment:
IIRC you're inlining the logo because the IDs cannot be duplicated, can you leave a comment about this?

Reply:
exactly. Im adding a comment